packages/esm-patient-allergies-app/src/allergies/allergies-form/allergy-form.component.tsx

```diff
@@ -160,30 +208,55 @@ function AllergyForm(props: DefaultWorkspaceProps) {
         }),
       };
       const abortController = new AbortController();
-      saveAllergy(payload, patientUuid, abortController)
-        .then(
-          (response: FetchResponse) => {
-            if (response.status === 201) {
-              mutate();
-              closeWorkspaceWithSavedChanges();
-              showSnackbar({
-                isLowContrast: true,
-                kind: 'success',
-                title: t('allergySaved', 'Allergy saved'),
-                subtitle: t('allergyNowVisible', 'It is now visible on the Allergies page'),
-              });
-            }
-          },
-          (err) => {
-            showSnackbar({
-              title: t('allergySaveError', 'Error saving allergy'),
-              kind: 'error',
-              isLowContrast: false,
-              subtitle: err?.message,
-            });
-          },
-        )
-        .finally(() => abortController.abort());
+      formContext === 'editing'
```

Comment:
It'd be better to extract the create and update logic into separate functions for readability.